cpp/inc/bond/core/detail/check_overflow.h

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#pragma once
+
+#include <limits>
+#include <stdexcept>
+#include <boost/static_assert.hpp>
+
+namespace bond
+{
+namespace detail
+{
+
+    inline void check_add_overflow(const char* ptr, uint32_t offset)
+    {
+        BOOST_STATIC_ASSERT(sizeof(const char *) == sizeof(std::uintptr_t));
```

Comment:
Can you add a comment saying that the numeric limit logic below requires that uintptr_t be exactly the same size as a pointer?

Reply:
OK.